lib/active_merchant/billing/gateways/monei.rb

```diff
@@ -225,6 +226,36 @@ def add_customer(xml, credit_card, options)
         end
       end
 
+      # Private : Convert ECI to ResultIndicator
+      # Possible ECI values:
+      # 02 or 05 - Fully Authenticated Transaction
+      # 00 or 07 - Non 3D Secure Transaction
+      # Possible ResultIndicator values:
+      # 01 = MASTER_3D_ATTEMPT
+      # 02 = MASTER_3D_SUCCESS
+      # 05 = VISA_3D_SUCCESS
+      # 06 = VISA_3D_ATTEMPT
+      # 07 = DEFAULT_E_COMMERCE
+      def eci_to_result_indicator(eci)
+        case eci
+        when '02', '05'
+          return eci
+        else
+          return '07'
+        end
+      end
+
+      # Private : Add the 3DSecure infos to XML
+      def add_three_d_secure(xml, options)
+        if options[:three_d_secure]
+          xml.Authentication(:type => '3DSecure') do
+            xml.ResultIndicator eci_to_result_indicator options[:three_d_secure][:eci]
+            xml.Parameter(:name => 'VERIFICATION_ID') { xml.text options[:three_d_secure][:cavv] }
+            xml.Parameter(:name => 'XID') { xml.text options[:three_d_secure][:xid] }
```

Comment:
won't 3DS 2.x be supported by MONEI?

Reply:
The current implementation of active_merchant connector with XML api only supports 3DS v1. We are planning to upgrade it to use our new JSON API that supports 3DS v2, but as there is a deadline for all connectors in september this is the simplest solution for now.

Reply:
Fair enough, thanks for the context.